server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -627,9 +627,11 @@ public VirtConsumerMap getHostConsumersMap(Owner owner, Iterable<String> hypervi
             .createAlias("owner", "o")
             .createAlias("hypervisorId", "hvsr")
             .add(Restrictions.eq("o.key", ownerKey))
-            .add(this.getHypervisorIdRestriction(hypervisorIds));
+            .add(this.getHypervisorIdRestriction(hypervisorIds))
+            .addOrder(Order.asc("hvsr.hypervisorId"));
 
-        return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria);
+        return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria)
+            .setLockMode(LockModeType.PESSIMISTIC_WRITE);
     }
 
     private Criterion getHypervisorIdRestriction(Iterable<String> hypervisorIds) {
```

Comment:
I'm unsure I follow. Are you suggesting that we replace the functionality of getHypervisorIdRestriction with your example?